### PR TITLE
DSD-1823: using buttonText in MultiSelect for CheckboxGroup legend value

### PR DESCRIPTION
**Pull request by EdwinGuzman**
Fixes JIRA ticket [DSD-1823](https://newyorkpubliclibrary.atlassian.net/browse/DSD-1823)

## This PR does the following:

- Uses the `MultiSelect`'s `buttonText` prop value as the internal `CheckboxGroup`'s label text value. The previous hardcoded value had no meaning, was repetitive, and was confusing. This updates gives it more context for screen readers.
- Also replace the internal `div` inside the accordion button to a `span` for better HTML DOM.

## How has this been tested?

- Locally in Storybook.

## Accessibility concerns or updates

- Fixes an issue found in Research Catalog.

### Accessibility Checklist

- [x] Checked Storybook's "Accessibility" tab for color contrast and other issues.
- [x] The feature works with keyboard inputs including tabbing back and forward and pressing space, enter, arrow, and esc keys.
- [x] For hidden text or when `aria-live` is used, a screenreader was used to verify the text is read.
- [x] For features that involve UI updates and focusing on DOM `ref`s, focus management was reviewed.
- [x] The feature works when the page is zoomed in to 200% and 400%.

## Open Questions

-

### Checklist:

- [ ] I have updated the Storybook documentation and changelog accordingly.
- [ ] I have added relevant accessibility documentation for this pull request.
- [ ] All new and existing tests passed.

### Front End Review:

- [ ] Review the Vercel preview deployment once it is ready.

[DSD-1823]: https://newyorkpubliclibrary.atlassian.net/browse/DSD-1823?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/components/MultiSelect/MultiSelect.tsx [hunk 1]
```diff
@@ -392,7 +393,7 @@ export const MultiSelect: ChakraComponent<
                 layout="column"
                 isFullWidth
                 isRequired={false}
-                labelText="Multi select checkbox group label"
+                labelText={buttonText}
```
Comment (bigfishdesign13) [inline src/components/MultiSelect/MultiSelect.tsx:396]: I feel that the button label text, on its own, is not descriptive enough here.  It simply mirrors what is already shown in the button.  What about adding `options` with it?

Ex: MultiSelect options

**Review by bigfishdesign13 — COMMENTED** [review]
It works and the need for the update makes sense, but maybe could be better.

File: src/components/MultiSelect/MultiSelect.tsx (hunk 1 above)
Comment (EdwinGuzman) [reply src/components/MultiSelect/MultiSelect.tsx:396]: Hmm yes, I added just the `buttonText` based on what Clare said but I'll ask for a review and ask if adding "options" is better.

Comment (EdwinGuzman) [reply src/components/MultiSelect/MultiSelect.tsx:396]: @bigfishdesign13 talked with Clare and she noted that the button text alone is fine, if anyone passes "Format options" as the button text, then we'll have a redundant legend text as "Format options options".

Comment (bigfishdesign13) [reply src/components/MultiSelect/MultiSelect.tsx:396]: Ok, then we're good.

**Review by bigfishdesign13 — APPROVED** [review]
Looks good.

**Review by dgcohen — APPROVED** [review]
Looks good!

**Pull request merged**